src/Dialogs/ExportDialog.vala

```diff
@@ -0,0 +1,226 @@
+        default_height = settings.export_height;
+
+        var sidebar_header = new Gtk.Grid ();
+        sidebar_header.vexpand = true;
+        sidebar_header.get_style_context ().add_class ("sidebar-export-header");
+        sidebar_header.height_request = 30;
+
+        var main_header = new Gtk.Grid ();
+        main_header.vexpand = true;
+        main_header.get_style_context ().add_class ("layers-panel");
+
+        var pane_header = new Gtk.Paned (Gtk.Orientation.HORIZONTAL);
+        pane_header.pack1 (sidebar_header, false, false);
+        pane_header.pack2 (main_header, false, false);
+        pane_header.get_style_context ().add_class ("export-titlebar");
+
+        // Hack to remove the default header area and replace it with
+        // a custom widget.
+        var header_area = get_header_bar ();
+        header_area.destroy ();
+        set_titlebar (pane_header);
+
+        // Another hack to remove the bottom action area.
+        var button = add_button ("OK", 3);
+        var button_area = button.get_parent ();
+        button_area.destroy ();
+
+        sidebar = new Gtk.Grid ();
+        sidebar.get_style_context ().add_class ("sidebar-export");
+        build_export_sidebar ();
+
+        var main = new Gtk.Grid ();
+        main.vexpand = true;
+        main.get_style_context ().add_class ("layers-panel");
+
+        var pane = new Gtk.Paned (Gtk.Orientation.HORIZONTAL);
+        pane.pack1 (sidebar, false, false);
+        pane.pack2 (main, true, false);
+
+        var content_area = get_content_area ();
+        content_area.border_width = 0;
+        content_area.add (pane);
+
+        pane.bind_property ("position", pane_header, "position");
+
+        pane_header.notify["position"].connect (() => {
+            pane_header.position = pane.position;
+        });
+
+        settings.bind ("export-paned", pane, "position", SettingsBindFlags.DEFAULT);
+
+        update_format_ui ();
+    }
+
+    private void build_export_sidebar () {
+        var grid = new Gtk.Grid ();
+        grid.expand = true;
+        grid.column_spacing = 12;
+        grid.margin_start = grid.margin_end = grid.margin_bottom = 12;
+        grid.row_spacing = 6;
+
+        // Folder location.
+        grid.attach (section_title (_("Export to:")), 0, 0, 1, 1);
+
+        folder_button = new Gtk.FileChooserButton (
+            _("Select Folder"),
+            Gtk.FileChooserAction.SELECT_FOLDER
+        );
+        folder_button.set_current_folder (
+            Environment.get_user_special_dir (UserDirectory.PICTURES)
+        );
+        folder_button.hexpand = true;
+        grid.attach (folder_button, 1, 0, 1, 1);
+
+        // File format.
+        var format_title = section_title (_("Format:"));
+        grid.attach (format_title, 0, 2, 1, 1);
+
+        file_format = new Gtk.ComboBoxText ();
+        file_format.append ("png", "PNG");
+        file_format.append ("jpg", "JPG");
+        //  file_format.changed.connect (update_format_ui);
```

Comment:
It seems that there's a bug in here, the first time the export dialog is opened, both the quaity and the compression scale are shown, even though the value of the `file_format` input is correctly populated from settings. 

Also, I think the `changed` event should be connected to `update_format_ui` (here it's commented). 
I tried to compile it without the comment and the switch works, but only after the first time I manually update the `file_format` field.